src/Nethermind/Nethermind.HonestValidator/ValidatorState.cs

```diff
@@ -28,6 +28,8 @@ public class ValidatorState
         private readonly Dictionary<BlsPublicKey, Slot> _attestationSlot = new Dictionary<BlsPublicKey, Slot>();
         private readonly Dictionary<BlsPublicKey, Slot> _proposalSlot = new Dictionary<BlsPublicKey, Slot>();
 
+        private readonly Dictionary<Slot, BlsPublicKey> _proposalDutyBySlot = new Dictionary<Slot, BlsPublicKey>();
```

Comment:
do they have to be concurrent?

Reply:
Maybe; is internal implementation, so encapsulated for easy change. 

Don't have multiple threads (yet) as it is all done async, with single check of duties, then proposal if needed (will only ever be one at a time). For attestation may have multiple attester threads, and can change then.